### DISCUSSION TITLE
How to make npm test working?

### DISCUSSION THREAD

**Discussion by tomasvladik** [eemeli/yaml#306] [Q&A] [answered]
Hello,

I wanted to reproduce an issue I have in the playground, but I'm not able to make tests working.
When I follow instructions from CONTRIBUTING.md :
```
git clone https://github.com/eemeli/yaml.git # or your own fork
cd yaml
git submodule update --init # required by tests; also fetches the docs & playground
npm install
npm test # just to be sure
```

it fails like this:
```

c:\_git\yaml>npm test

> yaml@2.0.0-7 test c:\_git\yaml
> jest --config config/jest.config.js

 FAIL  tests/doc/YAML-1.1.spec.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    C:\_git\yaml\tests\doc\YAML-1.1.spec.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import { source } from '../_utils'
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1517:14)
```

Is there anything else, that I need to set up? A specific version of node.js / npm or something?

**Comment by eemeli** [comment, marked as answer]
Nah, that _should_ work. And it does at least on MacOS and Linux. Perhaps some of the rules used in `config/jest.config.js` aren't matching because of the `\` directory separators that Windows uses instead of `/`? This seems like a pretty similar case: facebook/jest#7914.

As I don't have a local Windows machine to test on, I wonder if it'd be possible to use GitHub Actions to run the test suite also in a Windows environment?

**Reply by eemeli** [reply]
Hopefully fixed now by #307, though note that a checkout with `autocrlf true` in your git config will probably cause an error when running the yaml-test-suite tests. As debugging that is rather clunky from a non-Windows environment, I'm going to leave it for now as a problem for which I'd welcome a PR.

**Reply by tomasvladik** [reply]
Thanks, works really better now in Windows.
Tests are mostly passing, but somehow stuck in yaml-test-suite.js

![image](https://user-images.githubusercontent.com/64122091/132136353-af0d80be-9ccd-4577-9779-00429dbf2c08.png)

I will try to look at it tomorrow in debugger.

**Reply by tomasvladik** [reply]
yes, with core.autocrlf=false it is okay

**Reply by eemeli** [reply]
Thank you @tomasvladik for pointing out this problem. `autocrlf=true` should now also work, as this revealed a small number of corner-case bugs in handling CRLF content that have now been fixed in eee3cec, def167c and 23c7038.

**Reply by tomasvladik** [reply]
So, the issue we had in version 2.0.0-5 we don't see anymore in 2.0.0-8

It was freezing and memory leaking in a situation, when some extra new lines were at end of string literals.
Looks like some of your latest pullrequests has fixed it.
Thanks
T.
